the-collab-lab/tcl-73-smart-shopping-list
# User can share lists

## Description

Add functionality that allows a user to share a list from the `ManageList` page with another user by submitting this users email. This pull request includes validators to ensure that the user is logged in and the user they are sharing the list with has an account, and that the recipients email is a valid email format. 

## Related Issue

Closes #6

## Acceptance Criteria

  - [x] The ManageList view shows a form that allows the user to enter an email to invite an existing user to a list, in addition to the form that allows them to add items to that list.
  - [x] If the other user does not exist, the user is shown an error message that explains the problem

## Type of Changes

enhancement, accessibility 

## Updates

### Before

![Screenshot 2024-02-19 at 4 07 05 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/1c5c3898-fe9f-48ea-8e19-113e7335a88a)

### After
![Screenshot 2024-02-19 at 4 08 15 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/f75cf1ab-4196-4d70-9d56-9f310493bfc2)

![Screenshot 2024-02-19 at 4 10 01 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/a6c708dc-5e38-4e12-a9f3-57ad042d1e46)

![Screenshot 2024-02-19 at 4 14 20 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/965e8bd6-635a-406c-ae81-9e35ab0b4d20)

## Testing Steps / QA Criteria
- git pull
- git checkout pc-rs-feature-share-list
- npm ci
- npm start

- Navigate to the `ManageList` page.

Test for a successful submission:
- enter a valid email into the input field
- Mouse click on submit or press enter
- Watch for success alert

Test for a failed submission:
Invalid email
- Enter a invalid email (think no .com or @ symbol)
- Then mouse click on submit or press enter
- Watch for failure alert

No email entered
- Leave the input field empty
- Then mouse click on submit or press enter
- Watch for failure alert

User is not logged in and that the user owns the list
- Log out of app
- Navigate to the `ManageList` page
- Enter a valid email and then mouse click on submit or press enter
- Watch for failure alert

Check if the recipient is not in the db
- Enter a valid email into the input. Make sure this email isn't in the Cloud Firestore.
- Then mouse click on submit or press enter
- Watch for failure alert
